developer/js/source/errors.ts

```diff
@@ -0,0 +1,171 @@
+/**
+ * Log levels.
+ * 
+ * Note: Currently, this acts like a bit set, where the upper 4 bits of an
+ * unsigned 16 bit value are the log level flags.
+ */
+export enum LogLevel {
+  CERR_FATAL = 0x8000,
+  CERR_ERROR = 0x4000,
+  CERR_WARNING = 0x2000,
+  CERR_MEMORY = 0x1000,  // N.B., probably unused in TypeScript
+  CERR_INFO = 0x0000,  // N.B., not in widespread use
+};
+
+/**
+ * Error codes. Use these when logging messages.
+ * 
+ * Extends https://github.com/keymanapp/keyman/blob/99db3c0d2448f448242e6397f9d72e9a7ccee4b9/windows/src/global/inc/Comperr.h
+ */
+export enum KeymanCompilerError {
+  CERR_LEXICAL_MODEL_MIN = 0x0800,
+  CERR_LEXICAL_MODEL_MAX = 0x08FF,
```

Comment:
Can you add these two constants to windows/src/global/inc/comperr.h and windows/src/global/delphi/general/CompilerErrorCodes.pas please.